### PR TITLE
feat: java version of the application-load-balancer example

### PR DESCRIPTION
**Pull request by RomainMuller**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: java/application-load-balancer/src/main/java/software/amazon/awscdk/samples/ApplicationLoadBalancerStack.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        final VpcNetwork vpc = new VpcNetwork(this, "VPC");
+        final AutoScalingGroup asg = new AutoScalingGroup(this,
+                                                          "ASG",
+                                                          AutoScalingGroupProps.builder()
+                                                                               .withVpc(vpc)
+                                                                               .withInstanceType(new InstanceTypePair(InstanceClass.Burstable2,
+                                                                                                                      InstanceSize.Micro))
+                                                                               .withMachineImage(new AmazonLinuxImage())
+                                                                               .build());
```
Comment (sam-goodwin) [inline java/application-load-balancer/src/main/java/software/amazon/awscdk/samples/ApplicationLoadBalancerStack.java]: Is this really the indentation we want to encourage in our examples? I think it's terribly ugly.

How about:
```java
final AutoScalingGroup asg = new AutoScalingGroup(this,
    "ASG",
    AutoScalingGroupProps.builder()
        .withVpc(vpc)
        .withInstanceType(new InstanceTypePair(InstanceClass.Burstable2, InstanceSize.Micro))
        .withMachineImage(new AmazonLinuxImage())
        .build());
```

Comment (RomainMuller) [reply java/application-load-balancer/src/main/java/software/amazon/awscdk/samples/ApplicationLoadBalancerStack.java]: Fair enough ;)

**Comment by eladb** [comment]
@RomainMuller What's up with this?

**Comment by eladb** [comment]
I am going to close this for now. Reopen when ready.

**Pull request closed without merging**